### PR TITLE
Disable JMX bean creation by default

### PR DESCRIPTION
**Pull request by sdavids**
https://github.com/apache/logging-log4j2/issues/2462#issuecomment-2045821938

## Checklist

* Base your changes on `2.x` branch if you are targeting Log4j 2; use `main` otherwise
* `./mvnw verify` succeeds (if it fails due to code formatting issues reported by Spotless, simply run `./mvnw spotless:apply` and retry)
* Non-trivial changes contain an entry file in the `src/changelog/.2.x.x` directory
* Tests for the changes are provided
* [Commits are signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) (optional, but highly recommended)

### CODE REVIEW

**Comment by sdavids** [comment]
There are currently no tests for the JMX support.

`org.apache.logging.log4j.core.jmx.Server#isJmxDisabled` is `private` and `PropertiesUtil.getProperties()` lazily caches its values in a static field, i.e. on the JVM level.

So there are several options:

1. keep the status quo, i.e. no tests
2. make `org.apache.logging.log4j.core.jmx.Server#isJmxDisabled` package-private and have test along the lines of:
```java
@Tag("needsforkedvm")
class JmxEnablementTest {

  @ReadsSystemProperty
  @Test
  void isJmxDisabledDefault() {
   assertTrue(Server.isJmxDisabled());
  }

  @SetSystemProperty(key = "log4j2.disableJmx", value = "false")
  @Test
  void isJmxDisabledFalse() {
    assertFalse(Server.isJmxDisabled());
  }

  @SetSystemProperty(key = "log4j2.disableJmx", value = "true")
  @Test
  void isJmxDisabledTrue() {
    assertTrue(Server.isJmxDisabled());
  }
}
```
3. call `org.apache.logging.log4j.core.jmx.Server#reregisterMBeansAfterReconfigure` and and have test along the lines of:
```java
@Tag("needsforkedvm")
class JmxEnablementTest {

  @ReadsSystemProperty
  @Test
  void reregisterMBeansAfterReconfigureJmxSupportDefault() {
      Server.reregisterMBeansAfterReconfigure();
      final MBeanServer mbs = ManagementFactory.getPlatformMBeanServer();
      assertThat(mbs.getDomains()).doesNotContain("org.apache.logging.log4j2");
  }

  @SetSystemProperty(key = "log4j2.disableJmx", value = "false")
  @Test
  void reregisterMBeansAfterReconfigureJmxSupportEnabled() {
      Server.reregisterMBeansAfterReconfigure();
      final MBeanServer mbs = ManagementFactory.getPlatformMBeanServer();
      assertThat(mbs.getDomains()).contains("org.apache.logging.log4j2");
  }

  @SetSystemProperty(key = "log4j2.disableJmx", value = "true")
  @Test
  void reregisterMBeansAfterReconfigureJmxSupportDisabled() {
      Server.reregisterMBeansAfterReconfigure();
      final MBeanServer mbs = ManagementFactory.getPlatformMBeanServer();
      assertThat(mbs.getDomains()).doesNotContain("org.apache.logging.log4j2");
  }
```

Both 2. and 3. need a forked VM setup:


```xml
  <plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-surefire-plugin</artifactId>
    <configuration>
      <properties>
        <includeTags>needsforkedvm</includeTags>
      </properties>
      <forkCount>1</forkCount>
      <reuseForks>false</reuseForks>
    </configuration>
  </plugin>
```

https://junit.org/junit5/docs/current/user-guide/#writing-tests-tagging-and-filtering
https://maven.apache.org/surefire/maven-surefire-plugin/examples/fork-options-and-parallel-execution.html#forked-test-execution

**Comment by sdavids** [comment]
The double negative `log4j2.disableJmx=false` is unfortunate, but we would have to deprecate an existing system property and introduce a new one `log4j2.enableJmx` otherwise.

File: src/site/antora/modules/ROOT/pages/log4j-jmx-gui.adoc [hunk 1]
```diff
@@ -18,3 +18,10 @@ Licensed to the Apache Software Foundation (ASF) under one or more
 
 https://github.com/apache/logging-log4j-jmx-gui[The Apache Log4j JMX GUI] is a basic client GUI that can be used to monitor the `StatusLogger` output and to remotely modify the Log4j configuration.
 The client GUI can be run as a stand-alone application or as a JConsole plug-in.
+
+[NOTE]
+====
+The Java VM to be monitored has to enable Log4j JMX support, set the following system property when starting the Java VM:
+
+`log4j2.disableJmx=false`
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/log4j-jmx-gui.adoc:26]:
```suggestion
[source]
----
log4j2.disableJmx=false
----
```

**Review by vy — APPROVED** [review]
LGTM, great job @sdavids! :star_struck:

IIRC, @ppkarwasz had the unfortunate experience of working with JMX professionally. :sweat_smile: Hence, I would appreciate his review too.

**Comment by vy** [comment]
> There are currently no tests for the JMX support.
> ...
> 1. keep the status quo, i.e. no tests

I'd be in favor of this approach. I doubt if this feature has any significant usage at all. No need to complicate the code base and increase the build time (which is already excessive!) with tests of a hardly used feature.

**Review by ppkarwasz — APPROVED** [review]
LGTM too.

I don't think there is a need for tests either: since JMX support will not be present in `3.0.x`, we can add tests if we ever reintroduce it.

**Comment by ppkarwasz** [comment]
@sdavids,

Thank you for your contribution.

**Pull request merged**